src/deroff.rs

```diff
@@ -538,12 +538,311 @@ impl Deroffer {
         unimplemented!()
     }
 
+    fn quoted_arg(&self) -> bool {
+        unimplemented!()
+    }
+
+    fn request_or_macro(&mut self) -> bool {
+        // self.skip_char();
+        self.s = self.skip_char(&self.s, None).to_owned();
+
+        let s0 = &self.s[1..2];
+
+        match s0 {
+            "\\" => {
+                // if self.str_at(1) == Some('"') {
+                if Self::str_at(&self.s, 1) == "\"" {
+                    // self.condputs("\n");
+                    return true;
+                }
+            }
+            "[" => {
+                self.refer = true;
+                // self.condputs("\n");
+                return true;
+            }
+            "]" => {
+                self.refer = false;
+                // self.skip_char();
+                self.s = self.skip_char(&self.s, None).to_owned();
+                // return self.text();
+            }
+            "." => {
+                self.r#macro = false;
+                // self.condputs("\n");
+                return true;
+            }
+            _ => (),
+        };
+
+        self.nobody = false;
+        let s0s1 = self.s[..2].to_owned();
```

Comment:
Possible out of bound panic here, I believe we should use `self.s.get(..2).unwrap_or_default()` or something similar here.

Reply:
taken care of:
```rust
let s0s1 = self.s.chars().take(2).collect::<String>();
```

Reply:
Same as previous comment...is this actually fixed?